futurex_openedx_extensions/dashboard/views.py

```diff
@@ -1570,3 +1570,34 @@ def post(self, request: Any, *args: Any, **kwargs: Any) -> Response:
             {'url': upload_file(storage_path, file), 'uuid': short_uuid},
             status=http_status.HTTP_201_CREATED
         )
+
+
+@docs('CourseAssetsManagementView.create')
+@docs('CourseAssetsManagementView.list')
+@exclude_schema_for('retrieve', 'update', 'partial_update', 'destroy')
+class CourseAssetsManagementView(FXViewRoleInfoMixin, viewsets.ModelViewSet):  # pylint: disable=too-many-ancestors
+    """View to list and retrieve course assets."""
+    authentication_classes = default_auth_classes
+    permission_classes = [FXHasTenantAllCoursesAccess]
+    serializer_class = serializers.TenantAssetSerializer
+    pagination_class = DefaultPagination
+    fx_view_name = 'tenant_assets'
+    fx_default_read_write_roles = ['staff', 'fx_api_access_global']
+    fx_default_read_only_roles = ['staff', 'fx_api_access_global']
+    fx_allowed_write_methods = ['POST']
+    fx_allowed_read_methods = ['GET']
+    fx_view_description = 'api/fx/tenant/v1/assets/: Tenant Assets Management APIs.'
+    filter_backends = [DefaultOrderingFilter, DjangoFilterBackend, DefaultSearchFilter]
+    filterset_fields = ['tenant_id', 'created_by']
+    ordering = ['-id']
+    search_fields = ['file']
+
+    parser_classes = [MultiPartParser]
+
+    def get_queryset(self) -> QuerySet:
+        """Get the list of user uploaded files."""
+        if self.request.fx_permission_info['is_system_staff_user']:
+            return TenantAsset.objects.all()
```

Comment:
no need. `view_allowed_tenant_ids_full_access` will contains all tenants. Except if `tenant_ids` is passed it'll contain only that tenant
```suggestion
```

Reply:
yes.